c10/core/SymNodeImpl.h

```diff
@@ -158,6 +158,11 @@ class C10_API SymNodeImpl : public c10::intrusive_ptr_target {
   virtual double guard_float(const char* file, int64_t line) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual bool guard_size_oblivious(const char* file, int64_t line) {
+    // No improvement for unbacked SymBools by default, replace this
```

Comment:
Why cant we call into the impl you've added ?

Reply:
We will, but the SymNodeImpl.h is the superclass for all implementations of SymNode, including XLA's SymNode, which I just want them to guard.